File: C++ Fundamentals/Functions/Password Validator/Password Validator.cpp
#include <iostream>
using namespace std;

int countLetters(const string& text) {
	int result = 0;

	for (int i = 0; i < text.length(); i++) {
		if (isalpha(text[i])) result++;
	}

	return result;
}

int countDigits(const string& text) {
	int result = 0;

	for (int i = 0; i < text.length(); i++) {
		if (isdigit(text[i])) result++;
	}

	return result;
}

int main()
{
	string password;
	cin >> password;

	bool isValid = true;
	if (password.length() < 6 || password.length() > 10) {
		cout << "Password must be between 6 and 10 characters" << endl;
		isValid = false;
	}

	int letters = countLetters(password);
	int digits = countDigits(password);
	if (letters + digits != password.length()) {
		cout << "Password must consist only of letters and digits" << endl;
		isValid = false;
	}

	if (digits < 2) {
		cout << "Password must have at least 2 digits" << endl;
		isValid = false;
	}

	if (isValid) cout << "Password is valid" << endl;
}

File: JS Front-End/Functions and Statements/Exercises/password-validator.js
function solve(password) {
    const errors = [];
    if (password.length < 6 || password.length > 10) errors.push("Password must be between 6 and 10 characters");

    let letters = 0,
        digits = 0;
    for (const ch of password) {
        if ((ch >= "A" && ch <= "Z") || (ch >= "a" && ch <= "z")) letters++;
        else if (ch >= "0" && ch <= "9") digits++;
    }

    if (letters + digits !== password.length) errors.push("Password must consist only of letters and digits");
    if (digits < 2) errors.push("Password must have at least 2 digits");

    if (errors.length === 0) console.log("Password is valid");
    else {
        for (const error of errors) console.log(error);
    }
}

solve("logIn");
solve("MyPass123");
solve("Pa$s$s");
